src/Microsoft.DotNet.Interactive/Events/KernelEvent.cs

```diff
@@ -11,7 +11,17 @@ public abstract class KernelEvent
 {
     protected KernelEvent(KernelCommand command)
     {
-        Command = command ?? throw new ArgumentNullException(nameof(command));
+        if (command is null)
+        {
+            throw new ArgumentNullException(nameof(command));
+        }
+
+        if (command is AnonymousKernelCommand)
```

Comment:
I don't think this is valid. This API isn't opinionated about serialization concerns and there are valid use cases for non-serializable command types.

Reply:
Ah it is legal to have non-serializable commands in events? My assumption was that the system isn't opinionated about whether events flow in-process or out-of-process and that therefore non-serializable commands should be avoided in all events (to avoid cases where an event works fine when the target kernel is in-process - but fails when the same kernel is moved out-of-process).

Reply:
The design opinion is that out-of-process communication isn't required or assumed.

Reply:
Got it. I've reverted the change and also updated #2884 to capture what we discussed offline (i.e. when an internal / non-serializable command is encountered, perhaps we should only serialize the properties present on the base `KernelCommand` type + add better tests to catch cases where we inadvertently end up including internal commands in `KernelEvent` payloads).